### PR TITLE
AccessControl: Introduce new permissions to restrict access for reloading provisioning configuration

### PR DESCRIPTION
**Pull request by vtorosyan**
**What this PR does / why we need it**:

We have introduced new set of fine-grained permissions which allow users to restrict access for reloading provisioning configuration. This PR documents the new permissions and updates certain portion of docs to reflect a rename of one of the permissions related to the Grafana Enterprise.

**Which issue(s) this PR fixes**:

https://github.com/grafana/grafana-enterprise/issues/1648

### CODE REVIEW

**Review by gamab — APPROVED** [review]
Looks good to me, thanks for taking care of this :100:

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:42]:
```suggestion
| `provisioning:reload`      | `provisioners:*`                                                                        | Reload provisioning files. To find the exact scope for specific provisioner, see [Scope definitions]({{< relref "./permissions.md#scope-definitions" >}}). |
```

File: docs/sources/http_api/admin.md
Comment (osg-grafana) [inline docs/sources/http_api/admin.md:615]:
```suggestion
| Actions              | Scopes                      | Provision entities |
```

**Review by osg-grafana — CHANGES_REQUESTED** [review]

**Comment by osg-grafana** [comment]
Default to plural headings for consistency. :)

**Comment by vtorosyan** [comment]
@osg-grafana would appreciate second look at this. I have enabled auto-merge.

**Review by gamab — CHANGES_REQUESTED** [review]
Hey, went over this again, I think we are missing the fixed role description and grant :slightly_smiling_face:

**Review by gamab — APPROVED** [review]
Thanks, sorry for not seeing this earlier !

**Comment by vtorosyan** [comment]
@gamab thanks, good catch!

File: docs/sources/enterprise/access-control/fine-grained-access-control-references.md
Comment (achatterjee-grafana) [inline docs/sources/enterprise/access-control/fine-grained-access-control-references.md:18]:
```suggestion
| `fixed:provisioning:admin`     | `provisioning:reload`                                                                                                                                                                                                                                                        | Allows reload of provisioning configurations.                                                                                             |
```

File: docs/sources/http_api/admin.md
Comment (achatterjee-grafana) [inline docs/sources/http_api/admin.md:615]: Should be singular - Action, Scope, and Provision entity.

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

File: docs/sources/http_api/admin.md
Comment (osg-grafana) [reply docs/sources/http_api/admin.md:615]: Thank you for the pushback, @vtorosyan. I brought this to the Docs Squad last week to come to a collective decision. The result of which is that we can consistently advise folks on using singular as opposed to plural, by default.

**Comment by osg-grafana** [comment]
> Default to plural headings for consistency. :)

Nope. I brought this to the Docs Squad last week, and we have collectively decided to use singular column names.

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:61]: I am wondering about this verb and see that the code uses "read" and the plain text uses "get". What is happening here? Asked another way, what is the point of getting/reading a user from LDAP?

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:63]:
```suggestion
| `ldap.status:read`         | n/a                                                                                     | Verify the availability of the LDAP server or servers.                                                                                                                   |
```
I assume there can be one or more servers. Vet for accuracy.

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:65]: What is happening here? This sounds a bit cryptic. What is the user’s goal here?

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:66]: Please fill in the blank: "Read the _____ ?______ settings."

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:67]: Please fill in the blank: "Update the _____ ?______ settings."

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:80]:
```suggestion
| `services:accesscontrol` | Restrict an action to target only the fine-grained access control service. You can use this in conjunction with the `status:accesscontrol` actions.                                                                                               |
```

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:84]:
```suggestion
| `provisioners:*`         | Restrict an action to a set of provisioners. For example, `provisioners:*` matches any provisioner, and `provisioners:accesscontrol` matches the fine-grained access control [provisioner]({{< relref "./provisioning.md" >}}).                                 |
```

**Review by osg-grafana — CHANGES_REQUESTED** [review]

File: docs/sources/enterprise/access-control/permissions.md
Comment (gamab) [reply docs/sources/enterprise/access-control/permissions.md:65]: The goal is to know whether `fine grained access control` is in use. If the status says `disabled`, then, we are not enforcing fined-grained access control, but rather the legacy access control based on the organization role solely.

File: docs/sources/enterprise/access-control/fine-grained-access-control-references.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/fine-grained-access-control-references.md:36]: Sorry that it's hard to review, I've only added one more fixed permission to the list here and it shows as entire thing has changed.

Anyway, the proposal makes sense to me. 

The only thing is that `Grafana Admin` is the name of the role itself in [Grafana](https://grafana.com/docs/grafana/latest/permissions/organization_roles/#organization-admin-role), so Grafana administrator might not be something users are used to. That said, if this is how we want to go further, happy to make these changes.

File: docs/sources/enterprise/access-control/permissions.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/permissions.md:28]: I have not changed anything here, but I believe that this role gives access to list the roles and the response does not contain permissions.

File: docs/sources/enterprise/access-control/permissions.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/permissions.md:61]: There is currently [another conversation](https://raintank-corp.slack.com/archives/C01LU2Y5527/p1631179555006300) about consistency and naming the roles/permissions . Since this is not something changed in this PR, perhaps we can move that discussion there.

File: docs/sources/enterprise/access-control/permissions.md
Comment (vtorosyan) [reply docs/sources/enterprise/access-control/permissions.md:66]: Not sure I understand? This line has changed because of automatic formatting

**Comment by vtorosyan** [comment]
Thanks @osg-grafana for the review. Because of the automatic formatting, the PR seems much bigger than it is, as I changed only few things here (mainly adding new permissions/roles). I appreciate though the review of the rest, I have added bunch of your suggestions in the change and left some open, which we can hopefully address in a different conversation.

File: docs/sources/enterprise/access-control/permissions.md
Comment (mjseaman) [inline docs/sources/enterprise/access-control/permissions.md:60]:
```suggestion
| `org.users.role:update`    | `users:*`                                                                               | Update the organization role (`Viewer`, `Editor`, or `Admin`) of an organization.                                                                          |
```

File: docs/sources/enterprise/access-control/permissions.md
Comment (mjseaman) [reply docs/sources/enterprise/access-control/permissions.md:66]:
```suggestion
| `settings:read`            | `settings:*`<br>`settings:auth.saml:*`<br>`settings:auth.saml:enabled` (property level) | Read the [Grafana configuration settings]({{< relref "../../administration/configuration/_index.md" >}})                                                                                                                                       |
```

File: docs/sources/enterprise/access-control/permissions.md
Comment (mjseaman) [reply docs/sources/enterprise/access-control/permissions.md:67]:
```suggestion
| `settings:write`           | `settings:*`<br>`settings:auth.saml:*`<br>`settings:auth.saml:enabled` (property level) | Update any Grafana configuration settings that can be [updated at runtime]({{< relref "../../enterprise/settings-updates/_index.md" >}}).                                                                                                                                           |
```

**Comment by vtorosyan** [comment]
Thanks everyone for contributions. I'd appreciate if we can avoid increasing the scope of this PR. The automatic formatting resulted to many lines being shown as updated, however the scope of this PR includes only adding new permissions/roles. 

To be clear, I am not suggesting to leave the status-quo, rather let's give another go over docs and follow-up with more PRs. As we are updating docs in few places, the more we add here the harder it becomes to sync those changes together later .

File: docs/sources/enterprise/access-control/fine-grained-access-control-references.md
Comment (sakjur) [inline docs/sources/enterprise/access-control/fine-grained-access-control-references.md:37]: They're assigned to the strings `"Admin"` and `"Grafana Admin"` respectively. Do we have anything that makes it clear that that's when you want to change the organizational administrator's permissions, you should provision for `Admin` rather than `Administrator` or should that too be introduced? (not in this PR, regardless. We shouldn't let PRs be blocked by work that's not directly related to the PR)

Comment (vtorosyan) [reply docs/sources/enterprise/access-control/fine-grained-access-control-references.md:37]: The [provisioning docs](https://grafana.com/docs/grafana/latest/enterprise/access-control/provisioning/#built-in-role-assignments) and [API docs](https://grafana.com/docs/grafana/latest/http_api/access_control/#create-a-built-in-role-assignment) state this clearly. 

In other places it is referencing to the current docs where it's not very clear in either way. I'm not against this change (so I accepted the suggestion), however I agree that it won't make things less confusing for users. So if this is a one step forward towards consistent glossary and eliminates future work, I'm fine to update. But @osg-grafana would appreciate your call to unblock this PR, we can take this conversation to private and understand how we can move forward with FGAC docs.

Comment (sakjur) [reply docs/sources/enterprise/access-control/fine-grained-access-control-references.md:37]: That's good enough for me :+1: Thanks for the links

Comment (osg-grafana) [inline docs/sources/enterprise/access-control/fine-grained-access-control-references.md:37]:
```suggestion
| Admin          | `fixed:users:org:edit`<br>`fixed:users:org:read`<br>`fixed:reporting:admin:edit`<br>`fixed:reporting:admin:read`                                                                                                                                                                                                                                                                                                            | Allow access to the same resources and permissions that the [Organization Admin]({{< relref "../../permissions/organization_roles.md" >}}) has by default.                               |
```

**Review by osg-grafana — APPROVED** [review]
I think I successfully revert a couple of my original suggestions.

**Pull request merged**
